hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/map/IMapSqlConnector.java

```diff
@@ -152,6 +158,34 @@ public VertexWithInputConfig nestedLoopReader(
         return IMapJoiner.join(dag, name, toString(table), joinInfo, rightRowProjectorSupplier);
     }
 
+    @Override
+    public boolean supportsFullScanReader() {
+        return true;
+    }
+
+    @Nonnull
+    @Override
+    public Vertex fullScanReader(
+            @Nonnull DAG dag,
+            @Nonnull Table table0,
+            @Nullable Expression<Boolean> filter,
+            @Nonnull List<Expression<?>> projection) {
+        PartitionedMapTable table = (PartitionedMapTable) table0;
+        PlanNodeSchema schemaBefore = getScanSchemaBeforeProject(table);
+
+        JetMapScanMetadata mapScanMetadata = new JetMapScanMetadata(
+                table.getMapName(),
+                table.getKeyDescriptor(),
+                table.getValueDescriptor(),
+                getScanFieldPaths(table),
+                schemaBefore.getTypes(),
```

Comment:
`PlanNodeSchema` is created just to get types from it and be dropped immediately after... The other thing is, we have similar code in `FileTable` and `KafkaTable`, looks like such method would be useful on `Table`.